diff -u -r jflex-1.4.1/src/JFlex/Emitter.java jflex-1.4.1-patched/src/JFlex/Emitter.java
--- jflex-1.4.1/src/JFlex/Emitter.java	2004-11-07 13:12:46.000000000 +0100
+++ jflex-1.4.1-patched/src/JFlex/Emitter.java	2007-07-30 22:08:51.250000000 +0200
@@ -674,15 +674,24 @@
    * @prec  0 <= c <= 0xFFFF 
    */
   private void printUC(int c) {
-    if (c > 255) {
-      out.print("\\u");
-      if (c < 0x1000) out.print("0");
-      out.print(Integer.toHexString(c));
+    final String representation;
+    final int padding;
+    if (c <= 0xff) {
+        representation = Integer.toOctalString(c);
+        padding = 3;
+        out.print("\\");
+    } else {
+        representation = Integer.toHexString(c);
+        padding = 4;
+        out.print("\\u");
+    }
+
+    int len = representation.length();
+    while (len < padding) {
+        out.print("0");
+        len++;
     }
-    else {
-      out.print("\\");
-      out.print(Integer.toOctalString(c));
-    }    
+    out.print(representation);
   }
 
 
diff -u -r jflex-1.4.1/src/JFlex/PackEmitter.java jflex-1.4.1-patched/src/JFlex/PackEmitter.java
--- jflex-1.4.1/src/JFlex/PackEmitter.java	2004-04-12 20:07:48.000000000 +0200
+++ jflex-1.4.1-patched/src/JFlex/PackEmitter.java	2007-07-30 22:09:16.062500000 +0200
@@ -186,15 +186,24 @@
    * @param c the character to append
    */
   private void printUC(char c) {
-    if (c > 255) {
-      out.append("\\u");
-      if (c < 0x1000) out.append("0");
-      out.append(Integer.toHexString(c));
+    final String representation;
+    final int padding;
+    if (c <= 0xff) {
+        representation = Integer.toOctalString(c);
+        padding = 3;
+        out.append("\\");
+    } else {
+        representation = Integer.toHexString(c);
+        padding = 4;
+        out.append("\\u");
     }
-    else {
-      out.append("\\");
-      out.append(Integer.toOctalString(c));
+
+    int len = representation.length();
+    while (len < padding) {
+        out.append("0");
+        len++;
     }
+    out.append(representation);
   } 
 
   /**
